test/tests/functional/pbs_sched_preempt_enforce_resumption.py

```diff
@@ -101,20 +106,343 @@ def test_suspended_job_ded_time_calendared(self):
+
+        p = '"express_queue, normal_jobs, server_softlimits"'
+        a = {'preempt_prio': p}
+        self.server.manager(MGR_CMD_SET, SCHED, a, runas=ROOT_USER)
+
+        j1 = Job(TEST_USER1)
+        j1.set_attributes({ATTR_l + '.select': '1:ncpus=4',
+                           ATTR_l + '.walltime': 20})
+        jid1 = self.server.submit(j1)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid1)
+
+        j2 = Job(TEST_USER)
+        j2.set_attributes({ATTR_l + '.select': '1:ncpus=2',
+                           ATTR_q: 'expressq',
+                           ATTR_l + '.walltime': 8})
+        jid2 = self.server.submit(j2)
+        self.server.expect(JOB, {ATTR_state: 'S'}, id=jid1)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid2)
+
+        j3 = Job(TEST_USER)
+        j3.set_attributes({ATTR_l + '.select': '1:ncpus=2',
+                           ATTR_l + '.walltime': 10})
+        jid3 = self.server.submit(j3)
+        self.server.expect(JOB, {ATTR_state: 'S'}, id=jid1)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid2)
+        self.server.expect(JOB, {ATTR_state: 'Q'}, id=jid3)
+
+        self.server.expect(JOB, {ATTR_state: 'F'}, id=jid2, extend='x',
+                           offset=10, interval=2)
+        self.server.expect(JOB, {ATTR_state: 'S'}, id=jid1)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid3)
+
+        self.server.expect(JOB, {ATTR_state: 'F'}, id=jid3, extend='x',
+                           offset=15, interval=2)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid1)
+
+        p = '"express_queue, normal_jobs"'
+        a = {'preempt_prio': p}
+        self.server.manager(MGR_CMD_SET, SCHED, a, runas=ROOT_USER)
+
+        self.server.manager(MGR_CMD_UNSET, SERVER, 'max_run_res_soft.ncpus')
+
+    def test_preempted_job_queue_soft_limits(self):
+        """
+        This test confirms that a preempted job remains suspended if it has
+        violated queue soft limits
+        """
+        a = {ATTR_rescavail + '.ncpus': 4}
+        self.server.manager(MGR_CMD_SET, NODE, a, self.mom.shortname)
+
+        a = {'max_run_res_soft.ncpus': '[u:pbsuser1=2]'}
+        self.server.manager(MGR_CMD_SET, QUEUE, a, 'workq')
+
+        p = '"express_queue, normal_jobs, queue_softlimits"'
+        a = {'preempt_prio': p}
+        self.server.manager(MGR_CMD_SET, SCHED, a, runas=ROOT_USER)
+
+        j1 = Job(TEST_USER1)
+        j1.set_attributes({ATTR_l + '.select': '1:ncpus=4',
+                           ATTR_l + '.walltime': 20})
+        jid1 = self.server.submit(j1)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid1)
+
+        j3 = Job(TEST_USER)
+        j3.set_attributes({ATTR_l + '.select': '1:ncpus=2',
+                           ATTR_q: 'expressq',
+                           ATTR_l + '.walltime': 8})
+        jid2 = self.server.submit(j3)
+        self.server.expect(JOB, {ATTR_state: 'S'}, id=jid1)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid2)
```

Comment:
8 seconds is really short.  On the slower machines in our testbed, I can totally see that job 2 runs and finishes in the time job 3 is submitted.  The subsequent expect() on jid2 will fail.